include/circt/Dialect/LLHD/IR/ExtractOps.td

```diff
@@ -101,3 +101,118 @@ def LLHD_DextsOp : LLHD_Op<"dexts", [
     unsigned getTargetWidth() { return getLLHDTypeWidth(target().getType()); }
   }];
 }
+
+def LLHD_ExtfOp : LLHD_Op<"extf", [
+    NoSideEffect,
+    PredOpTrait<"'index' has to be smaller than the 'target' size",
+      CPred<"$index.cast<IntegerAttr>().getInt() < getTargetWidth()">>,
+    TypesMatchWith<"'result' type has to match type at 'index' of 'target', in "
```

Comment:
This seems awkwardly worded to me.  The description doesn't quite seem to match the predicate?